app/src/test/kotlin/com/wire/android/ui/authentication/create/username/CreateAccountUsernameViewModelTest.kt

```diff
@@ -0,0 +1,148 @@
+package com.wire.android.ui.authentication.create.username
+
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.ui.text.input.TextFieldValue
+import com.wire.android.navigation.BackStackMode
+import com.wire.android.navigation.NavigationCommand
+import com.wire.android.navigation.NavigationItemDestinationsRoutes
+import com.wire.android.navigation.NavigationManager
+import com.wire.android.util.EMPTY
+import com.wire.android.utils.CoroutineTestRule
+import com.wire.kalium.logic.NetworkFailure
+import com.wire.kalium.logic.feature.user.SetUserHandleUseCase
+import com.wire.kalium.logic.feature.auth.ValidateUserHandleUseCase
+import com.wire.kalium.logic.feature.user.SetUserHandleResult
+import io.mockk.MockKAnnotations
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.impl.annotations.MockK
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.TestCoroutineScheduler
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.amshove.kluent.shouldBe
+import org.amshove.kluent.shouldBeEqualTo
+import org.amshove.kluent.shouldBeInstanceOf
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+
+@OptIn(ExperimentalMaterialApi::class, ExperimentalCoroutinesApi::class)
+class CreateAccountUsernameViewModelTest {
+
+    @MockK private lateinit var navigationManager: NavigationManager
+    @MockK private lateinit var validateUserHandleUseCase: ValidateUserHandleUseCase
+    @MockK private lateinit var setUserHandleUseCase: SetUserHandleUseCase
+
+    private lateinit var createAccountUsernameViewModel: CreateAccountUsernameViewModel
+
+    @get:Rule
+    val coroutineTestRule = CoroutineTestRule()
```

Comment:
since we need coroutineTestRule

and MockKAnnotations.init(this) 

in every unit test using coroutine, maybe it is worth to create a base class and extend from it, instead of adding those thingies each time

Reply:
Sure, I created BaseTest (if you know a better name, let me know 😄 ).